### PR TITLE
Model read only simplified creation.

### PR DESCRIPTION
**Pull request by tlm**
We require the model metadata to available in the model database for services scoped to a model database. This required a lot of work from model creation components to duplicate data and handle the transfer from one database to another.

By exposing both databases to the service we can make the service do the heavy lifting and not push the complicated logic upwards. This allows for additions like agent version to much more simplified.

This commit also exposed the agent version in the model metadata.

The most controversial part of this PR is exposing both databases to the readonly model service. @SimonRichardson suggested this idea and it works well I think.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

1. Bootstrap a new controller and add a few models.
2. If this process succeeds with no errors we are working.

## Documentation changes

N/A

## Links

**Jira card:** JUJU-5342

### CODE REVIEW

File: domain/model/service/modelservice.go [hunk 1]
```diff
@@ -20,16 +21,23 @@ type ModelState interface {
 	Delete(ctx context.Context, uuid coremodel.UUID) error
 }
 
+// ModelsState represents the state required for reading all model information.
+type ModelsState interface {
```
Comment (wallyworld) [inline domain/model/service/modelservice.go:25]: ModelState (singular)

File: domain/model/service/modelservice.go [hunk 2]
```diff
@@ -20,16 +21,23 @@ type ModelState interface {
...
+	Get(context.Context, coremodel.UUID) (coremodel.Model, error)
+}
+
 // ModelService defines a service for interacting with the underlying model
 // state, as opposed to the controller state.
 type ModelService struct {
-	st ModelState
+	modelsST ModelsState
+	st       ModelState
```
Comment (wallyworld) [inline domain/model/service/modelservice.go:33]: Oh I see. Can we call it `ModelGetter` then. `ModelsState` is bleh IMO.

File: domain/model/service/modelservice.go [hunk 3]
```diff
@@ -20,16 +21,23 @@ type ModelState interface {
...
 }
 
 // NewModelService returns a new Service for interacting with a models state.
-func NewModelService(st ModelState) *ModelService {
+func NewModelService(modelsST ModelsState, st ModelState) *ModelService {
```
Comment (wallyworld) [inline domain/model/service/modelservice.go:37]: This then becomes `modelGetter ModelGetter` which is more intuitive.

File: domain/model/state/modelstate.go [hunk 4]
```diff
@@ -107,12 +108,15 @@ SELECT uuid,
 FROM model;
 `
 
-	var model coremodel.ReadOnlyModel
+	var (
+		controllerUUIDStr string
```
Comment (wallyworld) [inline domain/model/state/modelstate.go:112]: This makes me shudder, the 90s called and wants its Hungarian notation back.

File: domain/modelconfig/state/state.go [hunk 5]
```diff
@@ -95,6 +95,7 @@ func (st *State) ModelConfig(ctx context.Context) (map[string]string, error) {
 SELECT key,
        value
 FROM model_config
+UNION
```
Comment (wallyworld) [inline domain/modelconfig/state/state.go:98]: ????

**Review by wallyworld — APPROVED** [review]

File: domain/modelconfig/state/state.go (hunk 5 above)
Comment (tlm) [reply domain/modelconfig/state/state.go:98]: Fixed

File: domain/model/service/modelservice.go (hunk 1 above)
Comment (tlm) [reply domain/model/service/modelservice.go:25]: Done & Done

File: domain/model/service/modelservice.go (hunk 2 above)
Comment (tlm) [reply domain/model/service/modelservice.go:33]: Done & Done

File: domain/model/service/modelservice.go (hunk 3 above)
Comment (tlm) [reply domain/model/service/modelservice.go:37]: Done & Done

File: domain/model/state/modelstate.go (hunk 4 above)
Comment (tlm) [reply domain/model/state/modelstate.go:112]: Done & Done

File: domain/model/bootstrap/bootstrap.go [hunk 6]
```diff
@@ -101,30 +102,38 @@ func CreateModel(
...
-			return fmt.Errorf("missing model uuid")
-		}
-
-		var modelType coremodel.ModelType
-		err := controller.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
+		var m coremodel.Model
+		err := controllerDB.StdTxn(ctx, func(ctx context.Context, tx *sql.Tx) error {
 			var err error
-			modelType, err = state.GetModelType(ctx, tx, args.UUID)
+			m, err = state.Get(ctx, tx, id)
```
[the remark this answers is not in the document]
Comment (tlm) [reply domain/model/bootstrap/bootstrap.go:116]: Are you after `GetModel` ?

**Pull request merged**